### ISSUE TITLE
Asincify version of SQL.js (no COOP required!)

### ISSUE THREAD

**Issue by quolpr** [jlongster/absurd-sql#59]
I was able to achieve the same performance as `absurd-sql` has, but with asincify version of SQL.js(using `wa-sqlite`).

Here is the comparison:

https://user-images.githubusercontent.com/7958527/179220916-6d28aaa3-9ca4-453b-bccb-677aec516b83.mp4

And, you can also run some benchmarks at https://cerulean-bublanina-96bf3a.netlify.app/

The results are roughly the same, but no COOP is required anymore!

Here is the source code of backend https://github.com/trong-orm/wa-sqlite-web-backend/blob/main/src/IDBCachedWritesVFS.ts (it's a bit messy). I used the same tricks as `absurd-sql` do — don't write till transaction finish, use cursor if sequence read.

The problem now is that `wa-sqlite` is under GPL3 license, so we need someone who make MIT version of `wa-sqlite` 😅

**Comment by rhashimoto** [comment]
[wa-sqlite](https://github.com/rhashimoto/wa-sqlite) is _not_ a version of [SQL.js](https://github.com/sql-js/sql.js). No code is derived from SQL.js and it has a completely different API.

**Comment by quolpr** [comment]
Yep, you are right 🙂. Just want to point, that there is another way to achieve a good sqlite performance with asyncify version of SQLite
